File: api/reports/generate.php
<?php
// api/reports/generate.php - تولید گزارش روزانه
header('Content-Type: application/json; charset=utf-8');
header('Access-Control-Allow-Origin: *');
header('Access-Control-Allow-Methods: POST');

require_once $_SERVER['DOCUMENT_ROOT'] . '/config/database.php';
require_once $_SERVER['DOCUMENT_ROOT'] . '/includes/auth.php';
require_once $_SERVER['DOCUMENT_ROOT'] . '/includes/middleware.php';

if ($_SERVER['REQUEST_METHOD'] !== 'POST') {
    http_response_code(405);
    echo json_encode(['success' => false, 'message' => 'متد غیرمجاز']);
    exit;
}

try {
    $user_id = requireAuth();
    $input = json_decode(file_get_contents('php://input'), true);
    
    if (empty($input['unit']) || empty($input['date'])) {
        http_response_code(400);
        echo json_encode(['success' => false, 'message' => 'واحد و تاریخ الزامی است']);
        exit;
    }
    
    $unit = $input['unit'];
    $date = $input['date'];
    
    $database = new Database();
    $db = $database->getConnection();
    
    // دریافت اطلاعات کاربر
    $stmt = $db->prepare("SELECT report_prefix, report_suffix FROM users WHERE id = ?");
    $stmt->execute([$user_id]);
    $user = $stmt->fetch();
    
    // دریافت کارهای روز
    $sql = "SELECT t.*, 
                   CASE 
                       WHEN t.status = 'completed' THEN 'انجام شد'
                       WHEN t.status = 'in_progress' THEN 'در حال انجام'
                       WHEN t.status = 'stopped' THEN 'متوقف شد'
                       WHEN t.status = 'delegated' THEN 'ارجاع شد'
                       ELSE t.status
                   END as status_text
            FROM tasks t
            WHERE t.assignee_id = ? 
            AND (t.activity_section = ? OR t.activity_section IS NULL)
            AND (
                (t.task_type = 'periodic' AND t.due_date = ?)
                OR 
                (t.task_type = 'continuous' AND t.start_date <= ?)
            )
            ORDER BY t.priority DESC, t.created_at ASC";
    
    $stmt = $db->prepare($sql);
    $stmt->execute([$user_id, $unit, $date, $date]);
    $tasks = $stmt->fetchAll();
    
    // تولید متن گزارش
    $report_content = $user['report_prefix'] ? $user['report_prefix'] . "\n\n" : "";
    
    $report_content .= "گزارش کاری - تاریخ: " . date('Y/m/d', strtotime($date)) . "\n";
    $report_content .= "واحد: " . $unit . "\n\n";
    
    if (count($tasks) > 0) {
        $report_content .= "فعالیت‌های انجام شده:\n\n";
        
        foreach ($tasks as $index => $task) {
            $report_content .= ($index + 1) . ". " . $task['title'] . " - وضعیت: " . $task['status_text'] . "\n";
            if (!empty($task['description'])) {
                $report_content .= "   توضیحات: " . $task['description'] . "\n";
            }
        }
    } else {
        $report_content .= "هیچ کاری برای این تاریخ ثبت نشده است.\n";
    }
    
    $report_content .= "\n" . ($user['report_suffix'] ? $user['report_suffix'] : "");
    
    echo json_encode([
        'success' => true, 
        'content' => $report_content,
        'task_count' => count($tasks)
    ]);
    
} catch (Exception $e) {
    http_response_code(500);
    echo json_encode(['success' => false, 'message' => 'خطای داخلی سرور']);
}
?>

File: go-api/internal/reports/generate.go
package reports

import (
	"database/sql"
	"fmt"
	"net/http"
	"strings"
	"time"

	"bmp/go-api/internal/core"
)

// taskStatusLabels — پورتِ عیناً‌ همان TASK_STATUS_LABELS در
// includes/task-status-helper.php. هر تغییری آن‌جا باید این‌جا هم اعمال شود.
var taskStatusLabels = map[string]string{
	"not_started":           "شروع نشده",
	"in_progress":           "در حال انجام",
	"pending_approval":      "در انتظار تأیید",
	"completed":             "تکمیل شده",
	"approved":              "تأیید شده",
	"delegated":             "ارجاع شده",
	"rejected":              "متوقف شده(کارهای عادی)",
	"stopped":               "متوقف شده(فرآیندها)",
	"period_done":           "دوره انجام شد",
	"termination_requested": "در انتظار اتمام",
}

// Generate — پورتِ دقیقِ api/reports/generate.php
//
//	POST /go/api/reports/generate   body: {unit, date}
//	→ {"success":true,"content":"...","task_count":N}
//
// هیچ نوشتنی در دیتابیس ندارد — فقط متنِ گزارش را از رویِ کارهای همان روز
// می‌سازد (پیشوند/پسوندِ کاربر + فهرستِ کارها با برچسبِ فارسیِ وضعیت).
func Generate(db *sql.DB) http.HandlerFunc {
	const taskQ = `SELECT title, status, description
	               FROM tasks
	               WHERE assignee_id = ?
	                 AND (activity_section = ? OR activity_section IS NULL)
	                 AND (
	                     (task_type = 'periodic' AND due_date = ?)
	                     OR (task_type = 'continuous' AND start_date <= ?)
	                 )
	               ORDER BY priority DESC, created_at ASC`

	return func(w http.ResponseWriter, r *http.Request) {
		u := core.UserOf(r.Context())

		var body struct {
			Unit string `json:"unit"`
			Date string `json:"date"`
		}
		_ = core.ReadJSON(r, &body)
		if body.Unit == "" || body.Date == "" {
			core.WriteErr(w, http.StatusBadRequest, "واحد و تاریخ الزامی است")
			return
		}

		var prefix, suffix sql.NullString
		_ = db.QueryRow("SELECT report_prefix, report_suffix FROM users WHERE id = ?", u.ID).
			Scan(&prefix, &suffix)

		rows, err := db.Query(taskQ, u.ID, body.Unit, body.Date, body.Date)
		if err != nil {
			core.WriteErr(w, http.StatusInternalServerError, "خطای داخلی سرور")
			return
		}
		defer rows.Close()

		type taskRow struct{ title, status, description string }
		var tasks []taskRow
		for rows.Next() {
			var t taskRow
			var desc sql.NullString
			if err := rows.Scan(&t.title, &t.status, &desc); err != nil {
				core.WriteErr(w, http.StatusInternalServerError, "خطای داخلی سرور")
				return
			}
			t.description = desc.String
			tasks = append(tasks, t)
		}

		var b strings.Builder
		if prefix.Valid && prefix.String != "" {
			b.WriteString(prefix.String)
			b.WriteString("\n\n")
		}
		// معادلِ date('Y/m/d', strtotime($date)) در PHP.
		b.WriteString("گزارش کاری - تاریخ: ")
		b.WriteString(formatSlashDate(body.Date))
		b.WriteString("\n")
		b.WriteString("واحد: ")
		b.WriteString(body.Unit)
		b.WriteString("\n\n")

		if len(tasks) > 0 {
			b.WriteString("فعالیت‌های انجام شده:\n\n")
			for i, t := range tasks {
				label, ok := taskStatusLabels[t.status]
				if !ok {
					label = t.status
				}
				b.WriteString(fmt.Sprintf("%d. %s - وضعیت: %s\n", i+1, t.title, label))
				if t.description != "" {
					b.WriteString("   توضیحات: ")
					b.WriteString(t.description)
					b.WriteString("\n")
				}
			}
		} else {
			b.WriteString("هیچ کاری برای این تاریخ ثبت نشده است.\n")
		}

		b.WriteString("\n")
		if suffix.Valid {
			b.WriteString(suffix.String)
		}

		core.WriteJSON(w, http.StatusOK, map[string]any{
			"success": true, "content": b.String(), "task_count": len(tasks),
		})
	}
}

// formatSlashDate — معادلِ date('Y/m/d', strtotime($date)). اگر ورودی به
// شکلِ YYYY-MM-DD نبود (strtotime شکست می‌خورد)، همان رشتهٔ خام برگردانده
// می‌شود؛ نه ۱۹۷۰/۰۱/۰۱ی که strtotime(false) در PHP تولید می‌کرد.
func formatSlashDate(d string) string {
	t, err := time.Parse("2006-01-02", d)
	if err != nil {
		return d
	}
	return t.Format("2006/01/02")
}
